api_app/api/dependencies/database.py

```diff
@@ -1,60 +1,80 @@
 import logging
 from typing import Callable, Type
 
-from azure.cosmos import CosmosClient
-from azure.mgmt.cosmosdb import CosmosDBManagementClient
+from azure.cosmos.aio import CosmosClient
+from azure.mgmt.cosmosdb.aio import CosmosDBManagementClient
 from fastapi import Depends, FastAPI, HTTPException
 from starlette.requests import Request
 from starlette.status import HTTP_503_SERVICE_UNAVAILABLE
-
-from core import config, credentials
+from core import config
 from db.errors import UnableToAccessDatabase
 from db.repositories.base import BaseRepository
 from resources import strings
-
-
-def connect_to_db() -> CosmosClient:
+from azure.identity.aio import (
+    DefaultAzureCredential as DefaultAzureCredentialASync,
+    ManagedIdentityCredential as ManagedIdentityCredentialASync,
+    ChainedTokenCredential as ChainedTokenCredentialASync,
+)
+from azure.core.credentials import TokenCredential
+
+
+async def get_credential_async() -> TokenCredential:
+    """
+    Context manager which yields the default credentials.
+    """
+    managed_identity = config.MANAGED_IDENTITY_CLIENT_ID
+    credential = (
+        ChainedTokenCredentialASync(
+            ManagedIdentityCredentialASync(client_id=managed_identity)
+        )
+        if managed_identity
+        else DefaultAzureCredentialASync()
+    )
+    return credential
+
+
+async def connect_to_db() -> CosmosClient:
     logging.debug(f"Connecting to {config.STATE_STORE_ENDPOINT}")
 
     try:
-        primary_master_key = get_store_key()
+        database_credentials = await get_credential_async()
+        primary_master_key = await get_store_key(database_credentials)
 
         if config.STATE_STORE_SSL_VERIFY:
-            cosmos_client = CosmosClient(url=config.STATE_STORE_ENDPOINT, credential=primary_master_key, consistency_level='Session')
+            cosmos_client = CosmosClient(url=config.STATE_STORE_ENDPOINT, credential=primary_master_key)
         else:
             # ignore TLS (setup is a pain) when using local Cosmos emulator.
-            cosmos_client = CosmosClient(config.STATE_STORE_ENDPOINT, primary_master_key, consistency_level='Session', connection_verify=False)
```

Comment:
Just wondering, why consistency level setting to `Session` was removed here? 
do we want to rely on the account level consistency setting which is currently `Bounded Staleness`

Reply:
Great questions, I discussed it offline with @tamirkamara and the reason it was set to Session from the begging is simply because it was the default behavior of the sdk client. However, the async client does not seem to support 'Session', which I had to remove. I went and saw that the account level setting works, but if you have a better suggestion I would love your opinion since I am not very familiar with the subject.